unistd/pwd.c

```diff
@@ -24,69 +24,94 @@
 
 struct passwd pwnam;
 char pw_name[NAME_MAX];
+char intstr[16];
 char pw_gecos[128];
 char pw_dir[PATH_MAX];
 char pw_shell[PATH_MAX];
 char pw_passwd[128];
 
 
-struct passwd *getpwnam(const char *name)
+static void readpwentry(FILE *fp, char *pwentry, unsigned int maxlgth)
 {
-	FILE *fp;
-	char buf[1024];
+	char c = 0;
+	unsigned int i = 0;
+	c = fgetc(fp);
+	while (c != ':' && c != '\n' && c != EOF) {
```

Comment:
Maybe we would like to pass info about EOF outside, so user of this function can tell if unexpected EOF has occurred and give up on parsing file?

Reply:
Yea it would be good, but in the case of unexpected EOF there would be at most 6 failed `fgetc()` returning EOF before the
```
} while (!feof(fp))
```. 
would break the loop. Does this save justifies overdesigning this function?

Reply:
Ok, but if there's EOF and we're not expecting it, most likely file is corrupted and configuration is not valid.

Reply:
I do not think we can mitigate all `/etc/passwd` corruption scenarios without bloating this function to extreme size (i.e unexpected '/n' is now unhandled and will result in corruption of all records after it). 

It is a question whether we want to mitigate corruption scenarios **at all** if this file is meant to be modified only with root privilages